java/serving/src/test/java/feast/serving/it/ServingBenchmarkIT.java

```diff
@@ -0,0 +1,151 @@
+
+    registry.add("feast.stores[0].name", () -> "online");
+    registry.add("feast.stores[0].type", () -> "REDIS");
+    registry.add("feast.stores[0].config.host", () -> environment.getServiceHost("redis", 6379));
+    registry.add("feast.stores[0].config.port", () -> environment.getServicePort("redis", 6379));
+  }
+
+  protected ServingAPIProto.GetOnlineFeaturesRequest buildOnlineRequest(
+      int rowsCount, int featuresCount) {
+    List<ValueProto.Value> entities =
+        IntStream.range(0, rowsCount)
+            .mapToObj(i -> DataGenerator.createInt64Value(rand.nextInt(1000)))
+            .collect(Collectors.toList());
+
+    List<String> featureReferences =
+        IntStream.range(0, featuresCount)
+            .mapToObj(i -> String.format("feature_view_%d:feature_%d", i / 10, i))
+            .collect(Collectors.toList());
+
+    Map<String, ValueProto.RepeatedValue> entityRows =
+        ImmutableMap.of(
+            "entity", ValueProto.RepeatedValue.newBuilder().addAllVal(entities).build());
+
+    return TestUtils.createOnlineFeatureRequest(featureReferences, entityRows);
+  }
+
+  protected ServingAPIProto.GetOnlineFeaturesRequest buildOnlineRequest(int rowsCount) {
+    List<ValueProto.Value> entities =
+        IntStream.range(0, rowsCount)
+            .mapToObj(i -> DataGenerator.createInt64Value(rand.nextInt(1000)))
+            .collect(Collectors.toList());
+
+    Map<String, ValueProto.RepeatedValue> entityRows =
+        ImmutableMap.of(
+            "entity", ValueProto.RepeatedValue.newBuilder().addAllVal(entities).build());
+
+    return TestUtils.createOnlineFeatureRequest("benchmark_feature_service", entityRows);
+  }
+
+  @Test
+  public void benchmarkServing100rows10features() {
+    ServingAPIProto.GetOnlineFeaturesRequest req = buildOnlineRequest(100, 10);
+
+    measure(() -> servingStub.getOnlineFeatures(req), "100 rows; 10 features", 1000);
+  }
+
+  @Test
+  public void benchmarkServing100rows50features() {
+    ServingAPIProto.GetOnlineFeaturesRequest req = buildOnlineRequest(100, 50);
+
+    measure(() -> servingStub.getOnlineFeatures(req), "100 rows; 50 features", 1000);
+  }
+
+  @Test
+  public void benchmarkServing100rows100features() {
+    ServingAPIProto.GetOnlineFeaturesRequest req = buildOnlineRequest(100, 100);
+
+    measure(() -> servingStub.getOnlineFeatures(req), "100 rows; 100 features", 1000);
+  }
+
+  @Test
+  public void benchmarkServing100rowsFullFeatureService() {
+    ServingAPIProto.GetOnlineFeaturesRequest req = buildOnlineRequest(100);
+
+    measure(() -> servingStub.getOnlineFeatures(req), "100 rows; Full FS", 1000);
+  }
+
+  private void measure(Runnable target, String name, int runs) {
+    Stopwatch timer = Stopwatch.createUnstarted();
+
+    List<Long> records = Lists.newArrayList();
+
+    for (int i = 0; i < runs; i++) {
+      timer.reset();
+      timer.start();
+      target.run();
+      timer.stop();
+      if (i >= WARM_UP_COUNT) {
```

Comment:
seems fishy that you're doing your own warmups. Maybe consider JMH to manage the benchmarks?

Reply:
I spent an hour trying to make JMH annotations work together with junit in maven and I failed. I think it's not worth it. I wrote this code in a less time